### PR TITLE
bpo-38144: Add the root_dir and dir_fd parameters in glob.glob().

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue38144

### CODE REVIEW

File: Doc/library/glob.rst [hunk 1]
```diff
@@ -62,8 +71,11 @@ For example, ``'[?]'`` matches the character ``'?'``.
    .. versionchanged:: 3.5
       Support for recursive globs using "``**``".
 
+   .. versionchanged:: 3.9
```
Comment (ZackerySpytz) [inline Doc/library/glob.rst]: It should be `3.10`.

File: Doc/whatsnew/3.9.rst [hunk 2]
```diff
@@ -117,6 +117,14 @@ multiline indented output.
 (Contributed by Serhiy Storchaka in :issue:`37995`.)
 
 
+glob
```
Comment (ZackerySpytz) [inline Doc/whatsnew/3.9.rst]: This entry should be moved to `Doc/whatsnew/3.10.rst`.

File: Lib/test/test_glob.py [hunk 3]
```diff
@@ -53,6 +60,33 @@ def glob(self, *parts, **kwargs):
...
+            else:
+                expected = res
+            self.assertCountEqual([os.path.join(self.tempdir, x) for x in res2],
+                                  expected)
+            self.assertCountEqual(glob.iglob(pattern, **kwargs), res2)
+            bres2 = [os.fsencode(x) for x in res2]
+            self.assertCountEqual(glob.glob(os.fsencode(pattern), **kwargs), bres2)
+            self.assertCountEqual(glob.iglob(os.fsencode(pattern), **kwargs), bres2)
+
+        self.assertCountEqual(glob.glob(pattern, root_dir=self.tempdir, **kwargs), res2)
```
Comment (ZackerySpytz) [inline Lib/test/test_glob.py:80]: I think the code added in this PR should follow the PEP 8 style guide.

File: Lib/glob.py [hunk 4]
```diff
@@ -71,75 +81,120 @@ def _iglob(pathname, recursive, dironly):
...
+        return os.path.isdir(pathname)
+    try:
+        st = os.stat(pathname, dir_fd=dir_fd)
+    except (OSError, ValueError):
+        return False
+    else:
+        return stat.S_ISDIR(st.st_mode)
+
+def _join(dirname, basename):
+    # It is common if dirname or basename is empty
```
Comment (ZackerySpytz) [inline Lib/glob.py:193]: I don't think this comment is necessary.

Comment (serhiy-storchaka) [reply Lib/glob.py:193]: Why not? It explains why the special case was added.

Or maybe you can suggest better wording?

**Pull request merged**